sdks/python/apache_beam/io/gcp/dicomio.py

```diff
@@ -164,70 +167,109 @@ class DicomSearch(PTransform):
     }
 
   """
-  def __init__(self, credential=None):
+  def __init__(self, buffer_size=8, max_workers=5, credential=None):
     """Initializes DicomSearch.
     Args:
       credential: # type: Google credential object, if it is specified, the
       Http client will use it to create sessions instead of the default.
     """
+    self.buffer_size = buffer_size
+    self.max_workers = max_workers
     self.credential = credential
 
   def expand(self, pcoll):
-    return pcoll | beam.ParDo(_QidoSource(self.credential))
+    return pcoll | beam.ParDo(
+        _QidoSource(self.buffer_size, self.max_workers, self.credential))
```

Comment:
it may be good to rename this class to `_QuidoReadFn`? That's because Source is another class type in Beam.

Reply:
Changed